app/models/user.rb

```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  validates :name, presence: true, length: {maximum: 50}
```

Comment:
Small niggle: I think we put spaces before and after hash brackets:
```suggestion
  validates :name, presence: true, length: { maximum: 50 }
```

Reply:
Code climate will complain on DocFox's repo